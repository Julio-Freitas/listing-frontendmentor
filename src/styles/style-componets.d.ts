import { Theme } from './theme';

type Theme = typeof Theme;
declare module 'styled-components' {
  export interface DefaultTheme extends Theme {}
}
