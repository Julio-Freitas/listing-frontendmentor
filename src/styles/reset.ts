import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    // this is the shared style
    html, body {
    box-sizing: border-box;
    font-size: 62.5%;
    font-family: ${({ theme }) => theme.fonts.primary.leagueSpartan};
    background-color:${({ theme }) =>
      theme.colors.secondary.lightGrayishCyanBackground};
    margin: 0 auto;

    }

    #__next {
      display: flex;
      align-items: center;
      justify-content: center;
      flex-direction: column;
      max-width: 1140px;
      margin: 0 auto;
    }

    #__next main {
        display: flex;
        flex-direction: column;
        height: 100%;
        width: 100%;
    }
    * {
        box-sizing: border-box;
        margin: 0;
        padding: 0;
    }
    a{
        text-decoration: none;
    }

    input {
        all: unset;
    }

    button {
        cursor: pointer
    }
    img{
      display: inline-block;
      max-width:100%;
      margin:0;
    }
`;
