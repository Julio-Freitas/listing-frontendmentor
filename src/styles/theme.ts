export const Theme = {
  fonts: {
    primary: {
      leagueSpartan: 'League Spartan, sans-serif'
    },
    secondary: {
      lexendDeca: 'Lexend Deca, sans-serif'
    }
  },
  colors: {
    primary: {
      darkCyan: 'hsl(180, 29%, 50%)'
    },
    secondary: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilter: 'hsl(180, 31%, 95%)',
      darkGrayishCyanFilter: 'hsl(180, 8%, 52%)',
      veryDarkGrayinshCyan: 'hsl(180, 14%, 20%)'
    }
  }
};
