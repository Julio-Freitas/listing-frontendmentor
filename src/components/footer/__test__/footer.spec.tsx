import React from 'react';
import { render, screen } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Theme } from '../../../styles/theme';
import { Footer } from '..';

const renderWithProviderTheme = (component: React.ReactNode) => {
  return {
    ...render(<ThemeProvider theme={Theme}>{component}</ThemeProvider>)
  };
};

describe('<Card />', () => {
  it('should be render correctly', () => {
    const { container } = renderWithProviderTheme(<Footer />);
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByText('Powered by Julio')).toBeInTheDocument();
  });
});
