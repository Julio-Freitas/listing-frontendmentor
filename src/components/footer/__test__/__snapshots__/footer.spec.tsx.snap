// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should be render correctly 1`] = `
<footer>
  <a
    href="https://github.com/Julio-Freitas"
    rel="noopener noreferrer"
    target="_blank"
  >
    Powered by Julio
  </a>
</footer>
`;
